src/main/kotlin/com/dev_camp/reservation/service/ReservationServiceImpl.kt

```diff
@@ -0,0 +1,40 @@
+package com.dev_camp.reservation.service
+
+import com.dev_camp.auth.exception.InvalidAccessException
+import com.dev_camp.auth.exception.UserIdNotFoundException
+import com.dev_camp.common.exception.ApiException
+import com.dev_camp.common.exception.NotFoundException
+import com.dev_camp.reservation.domain.Reservation
+import com.dev_camp.reservation.domain.ReservationRepository
+import com.dev_camp.reservation.dto.ReservationDto
+import com.dev_camp.terrace.domain.Terrace
+import com.dev_camp.terrace.domain.TerraceRepository
+import com.dev_camp.terrace.domain.TerraceStatus
+import com.dev_camp.user.domain.User
+import com.dev_camp.user.domain.UserRepository
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+import javax.persistence.EntityNotFoundException
+
+@Service
+class ReservationServiceImpl (
+    private val reservationRepository: ReservationRepository,
+    private val userRepository: UserRepository,
+    private val terraceRepository: TerraceRepository
+) : ReservationService{
+    override fun createReservation(reservationDto: ReservationDto) : Reservation {
+        try {
+            val terrace: Terrace = terraceRepository.getById(reservationDto.terraceId)
+            val user: User = userRepository.getById(reservationDto.userId)
```

Comment:
try-catch를 아래처럼 없앨 수 있어요.

```suggestion
            val terrace: Terrace = terraceRepository.findById(reservationDto.terraceId).orElseThrow { TerraceIdNotFoundException() }
            val user: User = userRepository.findById(reservationDto.userId).orElseThrow { UserIdNotFoundException() }
```